TheHyruleCastAdventureGame.js

```diff
@@ -0,0 +1,536 @@
+    );
+    chamberChest();
+  } else if (choiceOne === "right") {
+    console.log(
+      chalk.redBright.italic(
+        `You walked up the right stairs and entered teh dinning hall.`
+      )
+    );
+    console.log(
+      chalk.redBright.italic(
+        `Inside the dining hall, you found a few powerfull dishes that you ate.`
+      )
+    );
+    console.log(
+      chalk.redBright.italic(
+        `You're full now, your attack and defense went up.`
+      )
+    );
+    console.log(chalk.redBright.italic(`There is 3 doors to leave the room.`));
+    console.log(
+      chalk.redBright.italic(
+        `One to the left, one to the right and one forward`
+      )
+    );
+    dinningHallDoors();
+  }
+};
+
+const dinningHallDoors = () => {
+  doorChoiceOne = readlineSync.question(
+    chalk.magenta.bold(`Which door do you want to go through ?\n`)
+  );
+  doorChoiceOne = doorChoiceOne.trim().toLowerCase();
+  if (doorChoiceOne === "left") {
+    dinningHallLeftDoor();
+  } else if (doorChoiceOne === "right") {
+    dinningHallRightDoor();
+  } else if (doorChoiceOne === "forward") {
+    dinningHallForwardDoor();
+  } else {
+    console.log(chalk.redBright.italic(`Please choose a correct answer`));
+    dinningHallDoors();
+  }
+};
+
+const dinningHallForwardDoor = () => {
+  console.log(
+    chalk.redBright.italic(
+      `You went through the door in front of you and walked into a room with only a huge set of spirling stairs.`
+    )
+  );
+  console.log(
+    chalk.redBright.italic(
+      ` You went up the stairs, and walked into the library on the third floor of the castle`
+    )
+  );
+  libraryFullPower();
+};
+const dinningHallLeftDoor = () => {
+  console.log(
+    chalk.redBright.italic(
+      `You walked out of the left door into the second floor hallway.`
+    )
+  );
+  secondFloorFullPower();
+};
+const dinningHallRightDoor = () => {
+  console.log(
+    chalk.redBright.italic(
+      `You opened and walked into a room with a slide that you took, and it kicked you out of the castle.`
+    )
+  );
+  kickedOutOfCastle();
+};
+
+const libraryFullPower = () => {
+  console.log(
+    chalk.redBright.italic(
+      `You're in the library, there's only one door to get out of the room.`
+    )
+  );
+  console.log(
+    chalk.redBright.italic(
+      `You tried opening the door but an inscription appeared in the door.`
+    )
+  );
+  console.log(
+    chalk.redBright.italic(
+      `Answer this question correctly and the door will open.`
+    )
+  );
+  array = ["Zelda", "Link", "Ganon"];
+  console.log(chalk.magenta.bold(`Who is the princess you're trying to save?`));
+  quizzanswer = readlineSync.keyInSelect(array);
+  playerChoice = array[quizzanswer];
+  if (playerChoice === array[0]) {
+    console.log(
+      chalk.redBright.italic(`The door opened and you walked through it`)
+    );
+    thirdFloor();
+  } else {
+    console.log(chalk.redBright.italic(`Wrong answer choose again.`));
+    libraryFullPower();
+  }
```

Comment:
This is also a good candidate for a while loop

Reply:
i changed this one.